### DISCUSSION TITLE
keycloak authentication middleware

### DISCUSSION THREAD

**Discussion by musicformellons** [tokio-rs/axum#856] [Q&A] [answered]
An example how to have this running with Keycloak jwt authentication. Maybe some middleware is needed? Something like: https://github.com/dsferruzza/actix-web-middleware-keycloak-auth

**Comment by davidpdrsn** [comment, marked as answer]
I'm not familiar with keycloak but we do have a JWT example. Porting what you're linking to to axum probably isn't hard, but should probably be a standalone crate.

**Comment by lpotthast** [comment]
Hey, for anyone looking: I have given this a shot over the weekend.

With Keycloak being one of if not the biggest player in the industry when it comes to authentication, I was surprised that there was still no crate for it. Guess the "mainstream" business-applications have yet to catch up to Rust.

Here is the crate: [https://crates.io/crates/axum-keycloak-auth](https://crates.io/crates/axum-keycloak-auth)
Always happy for feedback!

@davidpdrsn Should I prepare a PR providing an example for the axum repo?

**Reply by jplatte** [reply]
I don't think adding another example makes sense, instead you can open a PR linking to `axum-keycloak-auth` from [ECOSYSTEM.md](https://github.com/tokio-rs/axum/blob/main/ECOSYSTEM.md).

**Reply by davidpdrsn** [reply]
Amazing! The reason there isn't a crate is that no one built it yet 😅 so cool that you did!

I agree adding it to ECOSYSTEM.md is probably best.
